gammapy/cube/psf_kernel.py

```diff
@@ -0,0 +1,266 @@
+    that can be used to convolve `~gammapy.maps.WcsNDMap` objects.
+    It is usually computed from an EnergyDependentTablePSF
+
+    Parameters
+    ----------
+    psf_kernel_map : `~gammapy.maps.Map`
+        PSF kernel stored in a Map
+
+    Examples
+    --------
+
+    .. code:: python
+
+        import numpy as np
+        from gammapy.maps import WcsGeom, MapAxis
+        from astropy import units as u
+
+        # Define energy axis
+        energy_axis = MapAxis.from_edges(np.logspace(-1., 1., 4), unit='TeV', name='energy')
+
+        # Create WcsGeom
+        geom = WcsGeom.create(binsz=0.02*u.deg, width=10.0*u.deg, axes=[energy_axis])
+
+        # Extract EnergyDependentTablePSF from PSF IRF (here a PSF3D)
+        table_psf = psf.to_energy_dependent_table_psf(theta=0.5*u.deg)
+
+        psf_kernel = PSFKernel.from_table_psf(table_psf,geom, max_radius=1*u.deg)
+
+        some_map_convolved = psf_kernel.apply(some_map)
+    """
+
+    def __init__(self, psf_kernel_map):
+        self._psf_kernel_map = psf_kernel_map
+
+    @property
+    def psf_kernel_map(self):
+        return self._psf_kernel_map
+
+    @classmethod
+    def read(cls, *args, **kwargs):
+        """Read kernel Map from file."""
+
+        psf_kernel_map = WcsNDMap.read(*args, **kwargs)
+        return cls.from_map(psf_kernel_map)
+
+    @classmethod
+    def from_table_psf(cls, table_psf, geom, max_radius=None, factor=4):
+        """Create a PSF kernel from a TablePSF or an EnergyDependentTablePSF on a given MapGeom.
+
+        If the MapGeom is not an image, the same kernel will be present on all axes.
+
+        The PSF is estimated by oversampling defined by a given factor.
+
+        Parameters
+        ----------
+        table_psf : `~gammapy.irf.TablePSF` or ~gammapy.irf.EnergyDependentTablePSF`
+            the input table PSF
+        geom : `~gammapy.maps.WcsGeom`
+            the target geometry. The PSF kernel will be centered on the central pixel.
+            the geometry axes should contain an energy MapAxis.
+        max_radius : `~astropy.coordinates.Angle` or float
+            the maximum radius of the PSF kernel. If float, we assume unit is degree.
+        factor : int
+            the oversample factor to compute the PSF
+
+        Returns
+        -------
+        kernel : `~gammapy.cube.PSFKernel`
+            the kernel Map with reduced geometry according to the max_radius
+        """
+        # TODO : use PSF containment radius if max_radius is None
+        if max_radius is not None:
+            geom = _make_kernel_geom(geom, max_radius)
+
+        if isinstance(table_psf, TablePSF):
+            return cls(table_psf_to_kernel_map(table_psf, geom, factor))
+        else:
+            return cls(energy_dependent_table_psf_to_kernel_map(table_psf, geom, factor))
+
+    @classmethod
+    def from_gauss(cls, geom, sigma, max_radius=None, containment_fraction=0.99,
+                   normalize=True, factor=4):
```

Comment:
PyCharm shows: Unused options `normalize` and `factor`.
I think `normalize` should be removed (also from the docstring), and `factor` passed along.